pkg/v2/metrics/metrics.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	CurrentConnections       = "current_connections"
+	MaxConcurrentConnections = "max_concurrent_connections"
+	TotalConnections         = "total_connections"
+)
+
+var (
+	metricTypes = map[string]string{
+		CurrentConnections:       "gauge",
+		MaxConcurrentConnections: "gauge",
+		TotalConnections:         "counter",
+	}
+)
+
+type Metrics struct {
+	sync.RWMutex
+	metrics     map[string]int
+	fileWriters []*fileWriter
+}
+
+func New() *Metrics {
+	return &Metrics{
+		metrics:     make(map[string]int),
+		fileWriters: make([]*fileWriter, 0),
+	}
+}
+
+func (m *Metrics) WithFileWriter(filename string, format string, frequency time.Duration) *Metrics {
```

Comment:
i would prefer to interface this out (i'll merge anyways)

the writer should just be an interface instead of hard coded.  i wanted to make a writer back to authorized sockets

Reply:
No probs, can do. Sorry, misunderstood your comment about this.

Reply:
No probs, will do. Sorry, I misunderstood your earlier message about this.